shotover-proxy/src/message/mod.rs

```diff
@@ -313,27 +319,47 @@ pub enum Value {
     None,
     #[serde(with = "my_bytes")]
     Bytes(Bytes),
+    Ascii(String),
     Strings(String),
-    Integer(i64),
-    Float(f64),
+    Integer(i64, IntSize),
+    Double(f64),
+    Float(f32),
     Boolean(bool),
     Inet(IpAddr),
     List(Vec<Value>),
     Rows(Vec<Vec<Value>>),
     NamedRows(Vec<HashMap<String, Value>>),
     Document(HashMap<String, Value>),
     FragmentedResponse(Vec<Value>),
+    Set(Vec<Value>),
+    Map(Vec<(Value, Value)>),
```

Comment:
Why arent we using a HashMap? are cassandra maps ordered?

Reply:
Yeah we can use a HashMap, I think I had it has that because Cassandra gives it raw back like that but we can put it in a HashMap instead.

Reply:
Yeah I think a HashMap makes the most sense.
We can of course change it later if we realize we need something else.

Along the same line of thinking, we should also use HashSet for Set.

Reply:
Ah now I remember. We would need to implement `Hash` for all the types in Value.

Reply:
Doing something like that might mean we move "Collection" Values to other enum

Reply:
Oh, were we going to replace Document with Map?

Reply:
Oh I see, because the key is a Value instead of a String.

Reply:
What would it take to impl Hash for Value?

Reply:
you cant hash f64???
hmmm...

Reply:
Yeah that's why I ended up with the Vec's instead. Should we fine given we can trust that Cassandra will give us a valid map or set

Reply:
Hmmm its tempting to do something like:
```rust
Map(Vec<KeyValue>)

struct KeyValue {
    key: Value,
    value: Value,
}
```

however ... the ordering of key and then value is pretty usual, so maybe we dont need the struct.
We can always rewrite to use a struct if we find the code that uses it would benefit from it.

In general though, the `Map(Vec<(Value, Value)>)` approach is fine by me.

Reply:
this doesn't quite feel right to be honest from a Developer Experience perspective. 

I don't think it would be a ton of effort to implement hash for `Value`

Reply:
Implemented Hash for Value:
- switched the `HashMap`'s to `BTreeMap` which implements the `Hash` trait
- wrapped the floats in [`OrderedFloat`](https://docs.rs/ordered-float/latest/ordered_float/struct.OrderedFloat.html)

Reply:
Given that HashMap doesnt implement Hash BTreeMap seems like a reasonable compromise.

Set and Map variants need to be updated still.

Reply:
Fixed the rest of it. Should be good to resolve this now.